### PR TITLE
add dummy file name for some tiffseq

### PR DESCRIPTION
**Pull request by dylanmcreynolds**
This addresses a bug with the tiff sequence code.

When the files of a tiffsequence don't start with letters, just numbers, a hash was used for the key for that node. 

So:
```
tiff_files
  - 001.tiff
  - 002.tiff
  - a001.tiff
  - a002.tiff
  - b001.tiff
  - b002.tiff
```
would result in 

```
['d3a7b120-91f8-41ef-a1ff-f3e312f23525',
 'a',
 'b']
```

The unit test wasn't expecting a random name for this node, so it didn't test for it.

This PR detects that case and adds an `_` in as the key name. On a call with @taxe10 and @danielballan, we debated this default root name for only a couple seconds. I'm happy to hear better ideas.

### CODE REVIEW

**Comment by padraic-shafer** [comment]
This seems ok.  

Other thoughts, just to throw them out there: `''`, `' '`, or `None`.

```python 3
>>> d1 = {'': 0, 'a':1, 'b': 2}
>>> d1['']
0

>>> d2 = {' ': 0, 'a':1, 'b': 2}
>>> d2[' ']
0

>>> d3 = {None: 0, 'a':1, 'b': 2}
>>> d3[None]
0

# But not...
>>> d3[]
  File "<stdin>", line 1
    d3[]
       ^
SyntaxError: invalid syntax
```

**Comment by danielballan** [comment]
Since the keys also end up in path parameters, this unfortunately rules out a couple of those options:

* `''` ends up as `/api/v1/metadata/tif//` (emphasis on `//`) which HTTP elides into `/`.
* `None` has to to coerced to a string of some kind, so its special status as an out-of-band sentinel does not survive.
* Space technically works but may invite confusion.

I think we'll be happy if we stick with "The key must be a non-empty string."

**Comment by danielballan** [comment]
Feedback from @Wiebke, who thought about related issues in her work on #555, would be very welcome as well.

**Comment by dylanmcreynolds** [comment]
I almost feel like something super explicit might be better, even if a bit verbose:

```
['_unnamed_sequence',
 'a',
 'b']

```

**Comment by padraic-shafer** [comment]
Explicit and _brief_ would be even better: "_default", "_unnamed", "_blank", "_na", etc.

I favor including a non-alphanumeric to try to minimize name clashes with the user-defined sequences. Something that does not require url-encoding would be good. I suppose a leading "_" will list this key at the beginning for most sorting routines?

**Comment by danielballan** [comment]
:+1: for brevity. Any of these seem good to me:

> "_default", "_unnamed", "_blank", "_na", etc.

Using a number would get weird in this context:

```py
c['tif']['0'][0]  # first image
```

**Comment by dylanmcreynolds** [comment]
I committed `_unnamed`. I like it because it's pretty explicit.

**Review by danielballan — APPROVED** [review]
Looks good to me, but we should wait for @Wiebke to weigh in before we merge.

**Review by Wiebke — APPROVED** [review]
I think this is a sensible solution for unnamed sequences and I liked `_unnamed` as well.

**Pull request merged**
